Cannot do single `subparse()`

This is a simplified test case, showing that we can't do a single-element `subparse()`
```rust
#[derive(Ingot)]
pub struct Packet {
    pub bla: u8,

    #[ingot(subparse())]
    pub next_packet: InnerPacket, // works with `Repeated<InnerPacket>`
}

#[derive(Clone, Ingot)]
pub struct InnerPacket {
    pub baaaa: u8,
}
```

It fails with a cryptic message:
```
error[E0282]: type annotations needed
  --> src/lib.rs:76:10
   |
76 | #[derive(Ingot)]
   |          ^^^^^ cannot infer type
   |
   = note: this error originates in the derive macro `Ingot` (in Nightly builds, run with -Z macro-backtrace for more info)
```

--- a/ingot-macros/src/packet/mod.rs
+++ b/ingot-macros/src/packet/mod.rs
@@ -1783,7 +1783,7 @@ impl StructParseDeriveCtx {
                     use ::ingot::types::ParseChoice;
                     use ::ingot::types::HeaderParse;
 
-                    let mut hint = None;
+                    let mut hint: Option<<Self as NextLayer>::Denom> = None;
 
                     #( #segment_fragments )*
 

